### PR TITLE
Update steady state timing to show median time

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Also, use high res stopwatch timing if available, and emit events for
various phases of the test so we can explore the phase behavior of the
test more easily.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
PTAL @rynowak, @noahfalk, @jorive

File: src/MusicStore/Program.cs [hunk 1]
```diff
@@ -59,33 +73,49 @@ public static void Main(string[] args)
...
+                    double averageResponse = 1000 * ((double)responseTimes.Sum() / N / Stopwatch.Frequency);
+                    double medianResponse = 1000 * ((double)responseTimes.OrderBy(t => t).ElementAt(N / 2) / Stopwatch.Frequency);
+                    Console.WriteLine("Steadystate min response time: {0:F2}ms", (1000 * minRequestTime) / Stopwatch.Frequency);
+                    Console.WriteLine("Steadystate max response time: {0:F2}ms", (1000 * maxRequestTime) / Stopwatch.Frequency);
+                    Console.WriteLine("Steadystate average response time: {0:F2}ms", averageResponse);
+                    Console.WriteLine("Steadystate median response time: {0:F2}ms",  medianResponse);
+                }
+                else
+                {
+                    long averageResponse = responseTimes.Sum() / N;
```
Comment (noahfalk) [inline src/MusicStore/Program.cs]: Do you know what platforms don't have the HighFrequency counter available? If the case effectively never happens on platforms we care about then we don't have to worry about. If it does show up though...

I'd expect the average and median will generally be an integer between 0-3 which is so course its hard to measure any improvement/regression. What I've done in the past give up on measuring individual requests and instead measure batches of 1000 requests. It doesn't provide a good response distribution but it does give an average with enough precision that you can measure changes.

File: src/MusicStore/Program.cs [hunk 2]
```diff
@@ -59,33 +73,49 @@ public static void Main(string[] args)
...
                 var minRequestTime = long.MaxValue;
                 var maxRequestTime = long.MinValue;
-                var averageRequestTime = 0.0;
+                int N = 1001;
+                long[] responseTimes = new long[N];
 
-                Console.WriteLine("Running 100 requests");
-                for (var i = 1; i <= 100; i++)
+                Console.WriteLine($"Running {N} requests");
+                JitBenchEventSource.Log.BeginRequests(N);
```
Comment (noahfalk) [inline src/MusicStore/Program.cs]: In the past I think I've observed the warmup period can extend for a few thousand requests before the app really seems to settle into a long term average. You may want to issue a large number of unmeasured requests, or separately measure a pool or warmup requests and a pool of steady-state requests.

File: src/MusicStore/Program.cs (hunk 1 above)
Comment (AndyAyersMS) [reply src/MusicStore/Program.cs]: I don't know which platforms are lacking high frequency counters. Let me see if I can find out.

The whole point of using the HF counters is to get at some fractional digits. So in the HF case we'll see median response times like 3.193ms.

File: src/MusicStore/Program.cs (hunk 2 above)
Comment (AndyAyersMS) [reply src/MusicStore/Program.cs]: I'll play with this a bit more -- I can simply wrap this loop in another and either print each batches' data or else just print the last one.

File: src/MusicStore/Program.cs (hunk 1 above)
Comment (noahfalk) [reply src/MusicStore/Program.cs]: Yeah I definitely like your proposal to use HF when available, I just wasn't sure how much we needed care about the non-HF case.

**Comment by rynowak** [comment]
@AndyAyersMS 

LGTM to me in general. Let me know when you're ready to merge

**Comment by AndyAyersMS** [comment]
Updated per Noah's feedback. Steady state times seen by the CoreCLR jit bench harness seem pretty stable now (it will just use the last set of numbers).

**Comment by noahfalk** [comment]
LGTM

**Comment by AndyAyersMS** [comment]
@rynowak this is ready to merge

**Pull request merged**